core/src/main/java/hudson/model/FileParameterDefinition.java

```diff
@@ -99,14 +99,22 @@ private String getFileName(String possiblyPathName) {
         return possiblyPathName;
     }
 
+    @SuppressWarnings("deprecation")
     @Override
     public ParameterValue createValue(CLICommand command, String value) throws IOException, InterruptedException {
         // capture the file to the server
-        FilePath src = new FilePath(command.checkChannel(),value);
         File local = File.createTempFile("jenkins","parameter");
-        src.copyTo(new FilePath(local));
+        String name;
+        if (value.isEmpty()) {
+            FileUtils.copyInputStreamToFile(command.stdin, local);
```

Comment:
> I would also handle `null`

It cannot be null (nor did the original code path check for that).

> and blank strings

That is what we are checking for here.

> It will work if and only if there is one file parameter in the build, right?

Correct, as noted in the help text.

> So for multiple params usage of old CLI is the only option?

Yes, though you could simply define a single file parameter as a ZIP with defined entries.